plugin/src/main/java/com/playmonumenta/scriptedquests/timers/CommandTimer.java

```diff
@@ -21,16 +22,27 @@
 	private final TreeMap<TimerCoords, ArmorStand> mCoords = new TreeMap<>();
 
 	private final Plugin mPlugin;
+	private final ArrayList<LinkedHashMap<UUID, CommandTimerInstance>> mTickTimers;
 	private final Map<UUID, CommandTimerInstance> mTimers;
 	private final BukkitRunnable mRunnable;
 	private final int mPeriod;
 	private final String mPeriodStr;
+	private int mCounter = 0;
+	private int mScheduler = 0;
 
 	public CommandTimer(Plugin plugin, int period) {
 		mPlugin = plugin;
 		mPeriod = period;
+		//List of timers sorted in their scheduled tick
+		mTickTimers = new ArrayList<LinkedHashMap<UUID, CommandTimerInstance>>();
```

Comment:
When creating the list, add the size to the constructor here - (period). This sizes the list correctly, so it won't need to be resized later. Not a big deal, but since you know the size ahead of time might as well.